src/resolver.cpp

```diff
@@ -235,211 +222,95 @@ std::string UsdOpenAssetIOResolver::_CreateIdentifier(
           // resolve to an absolute path, making the anchorAssetPath redundant
           // (for now).
           // TODO(DF): Allow the manager to provide an identifier representing
-          //  an "anchored" entity reference.
+          //  an "anchored" entity reference via getWithRelationship.
           identifier = assetPath;
         } else {
-          identifier = ArDefaultResolver::_CreateIdentifier(
-              assetPath,
-              locationInManagerContextForEntity(manager_, readContext_, anchorAssetPath));
+          identifier = ArDefaultResolver::_CreateIdentifier(assetPath, anchorAssetPath);
         }
 
         return identifier;
       },
-      logger_, fnName);
-
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] "
-           << "result: " << result;
-    logger_->debug(logMsg.str());
-  }
-  return result;
-}
-
-// TODO(DF): Implement for publishing workflow.
-std::string UsdOpenAssetIOResolver::_CreateIdentifierForNewAsset(
-    const std::string &assetPath, const ArResolvedPath &anchorAssetPath) const {
-  auto result = ArDefaultResolver::_CreateIdentifierForNewAsset(assetPath, anchorAssetPath);
-  logger_->debug(TF_FUNC_NAME());
-
-  logger_->debug("  assetPath: " + assetPath);
-  logger_->debug("  anchorAssetPath: " + anchorAssetPath.GetPathString());
-  logger_->debug("  result: " + result);
-
-  return result;
+      logger_, TF_FUNC_NAME());
 }
 
 ArResolvedPath UsdOpenAssetIOResolver::_Resolve(const std::string &assetPath) const {
-  const std::string fnName = TF_FUNC_NAME();
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] " << fnName
-           << "\n  assetPath: " << assetPath;
-    logger_->debug(logMsg.str());
-  }
-
-  auto result = catchAndLogExceptions(
+  return catchAndLogExceptions(
       [&] {
-        if (manager_->isEntityReferenceString(assetPath)) {
-          // TODO(DF): We may wish to do more here, e.g.
-          //  `finalizedEntityVersion()`
-          return ArResolvedPath{assetPath};
+        if (auto entityReference = manager_->createEntityReferenceIfValid(assetPath)) {
+          return ArResolvedPath{resolveToPath(manager_, readContext_, entityReference.value())};
```

Comment:
[consider][minor] `.value()` is a safe check that throws an exception if the `std::optional` is not set, but we know it's set because of the enclosing `if`, so we can use the slightly more performant `*` operator.
```suggestion
          return ArResolvedPath{resolveToPath(manager_, readContext_, *entityReference)};
```